markup/forms.md

```diff
@@ -0,0 +1,71 @@
+# Forms
+
+## Checklist
+
+### Buttons rather than inputs
+
+- Specify the required attributes (name and value)
+
+### Multistep forms over multiple submit buttons
+
+### Server-side Validation
+
+- Start with server-side validation
+- Return an error message for each erroring form element.
+
+### Client-side validation
+
+### Accessibility
+
+### Label every input
+
+### Use the most simple form control that does the job
+
+### Minimise the number of fields to fill out
+
+### Mark optional fields
+
+### Size according to the expected input
+
+### Retain focus for every input element
+
+- For preference, enhance it! Make it part of the brand!
+
+### Don't ask users to repeat their email address
+
+### Provide a "show password" option
+
+### Don't split data fields
+
+### Avoid dropdown menus
+
+### Provide examples of input
+
+- Don't use placeholders!
+
+### Offer help text
+
+### Masked input
+
+### Use appropriate input types
+
+### Explain clearly why you need any private information
+
+### Use autocomplete when useful
+
+### Enable auto-filling of personal details
+
+### Use fieldsets and legends
+
+### Make use of native validation, enhance when necessary
+
+### Avoid native datepickers
+
+- Controversial.
+- Often better to stick with date fields.
+
+## Resources
+
+- https://www.smashingmagazine.com/2018/08/best-practices-for-mobile-form-design/
+- https://github.com/cferdinandi/bouncer
+- https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/forms/Basic_form_hints
```

Comment:
- https://adamsilver.io/articles/form-design-from-zero-to-hero-all-in-one-blog-post/